### ISSUE TITLE
Update Sentinel2 driver for Major Products Upgrade Upcoming (baseline 04.00)

### ISSUE THREAD

**Issue by tfeuvrier-cs** [OSGeo/gdal#5169]
## Why ?
ESA is pleased to announce that the new Sentinel-2 Processing Baseline 04.00 will be deployed on 25 January 2022.
The goal of this issue is to update the GDAL Sentinel-2 driver to be compliant with the new product Sentinel-2 baseline tagged 04.00.

## References
Major Products Upgrade Upcoming are described here https://sentinel.esa.int/web/sentinel/-/copernicus-sentinel-2-major-products-upgrade-upcoming
The reference document "Sentinel-2 Products Specification Document has been update " https://sentinel.esa.int/documents/247904/685211/Sentinel-2-Products-Specification-Document-14_8.pdf

## Dataset for validation 
Sentinel-2 Level-1C and Level-2A products of the new processing baseline, along with the corresponding products of the current processing baseline, are available for browsing and download from https://colhub2.copernicus.eu/s2tds (https://sentinel.esa.int/web/sentinel/-/deployment-of-sentinel-2-processing-baseline-04.00) 

The autotest sentinel2.py will be updated to take into account thare update Data use to validate the driver from "Deployment of Sentinel-2 Processing Baseline 04.00 and Provision of New Sample Products" 

## Operating system

Ubuntu 16.04 64 bit

**Comment by rouault** [comment]
@tfeuvrier-cs Are you working on this update ?

**Comment by tfeuvrier-cs** [comment]
Yes, I started a little and I will continue this task but only when I return from vacation on Monday February 14.

**Comment by rouault** [comment]
is this issue still current ?

**Comment by rouault** [comment]
@tfeuvrier-cs any update on this ?

**Comment by rouault** [comment]
closing due to lack of feedback from reporter
